### DISCUSSION TITLE
Native DLL Dependency Excluded From Managed Custom Action

### DISCUSSION THREAD

**Discussion by claybrooks** [wixtoolset/issues#7833] [Questions]
Wix v4

My managed custom action has nuget package references which includes Microsoft.Data.SqlClient.SNI.x86.dll.  Microsoft.Data.SqlClient.SNI.x86.dll is a native dll and seems to be getting skipped by the utility that generates the *.CA.dll.

In order to work around this issue, I've copied Microsoft.Data.SqlClient.SNI.x86.dll to a location and added it as a content file with "Copy Always" enabled to my custom action project.  This seems to get it packaged with the installer.

Is there a better way to do this?  It seems like native dlls, especially when it's a transitive dependency, [is an issue for SDK style projects in general.  ](https://learn.microsoft.com/en-us/sql/connect/ado-net/sqlclient-troubleshooting-guide?view=sql-server-ver15#unable-to-load-native-sni-server-name-indication-library)

**Comment by chrpai** [comment]
Pre nuget what your doing was the way.  I'd say write this up as an issue and see if there's anything the WiX team can do to auto include these.
